### PR TITLE
Add best practice for documenting CSS architecture

### PR DESCRIPTION
**Pull request by tysongach**
- Each project is unique and our guides on front-end development
purposefully do not prescribe a specific way to structure and
organize CSS.
- Documenting how the project's CSS is structured and organized allows
for a shared understanding, making it easier to contribute to
and review.

### CODE REVIEW

**Comment by gylaz** [comment]
Where should one document this? Inline comments? The README? Worth adding that specification to the PR, or doesn't matter?

**Comment by tysongach** [comment]
@gylaz That’s a great question, and I debated specifying a location. I usually place this information within the readme, but I also didn’t necessarily enforce a location. I guess it should belong where any other documentation goes; that could be a readme, or a myriad of other places. I’m open to ideas here…

**Comment by gylaz** [comment]
The README could work. Wasn't sure if there was some more appropriate place for front-end. Hmm, how about the main manifest file?

**Comment by tysongach** [comment]
@gylaz After thinking about it for a bit, I think the README would work best. Although the main manifest file is more contextually appropriate, the README is in Markdown and allows for rich formatting and easy linking to external resources. Doing that within Sass comments can be messy.

I’ve added a note about this in 5325a9012bdf990565b1f056bed54e9b9769ccc9.

**Comment by creuter** [comment]
👍

File: best-practices/README.md [hunk 1]
```diff
@@ -219,6 +219,12 @@ HTML
 CSS
 ---
 
+* Document the project's CSS architecture in the README, including things
+  such as:
+  * Organization of stylesheet directories and Sass partials
+  * Selector naming
```
Comment (jasonramirez) [inline best-practices/README.md]: Selector naming convention?

File: best-practices/README.md [hunk 2]
```diff
@@ -219,6 +219,12 @@ HTML
...
+  * Browser support
+  * Code linting
```
Comment (jasonramirez) [inline best-practices/README.md]: Can you be more specific? As in "What tools you are using for code linting" or "The library that is being used for code linting"?

File: best-practices/README.md (hunk 1 above)
Comment (tysongach) [inline best-practices/README.md]: Updated in 6e61ae39e8d58fff6ea42e8ca4255dd0e648091d.

File: best-practices/README.md (hunk 2 above)
Comment (tysongach) [inline best-practices/README.md]: Updated in 6e61ae39e8d58fff6ea42e8ca4255dd0e648091d.

**Comment by jasonramirez** [comment]
🍔  all the way

**Comment by tysongach** [comment]
I expanded on the list of possible places this documentation could go, to include component libraries and style guides.

**Comment by jasonramirez** [comment]
@tysongach great. What is keeping you from merging this in? Are you waiting for more time to pass?

**Comment by tysongach** [comment]
@jasonramirez These are guides for our team as a whole, so I like to let them sit for just a bit longer and get people's feedback. I've been popping the PR in the #design Slack channel each Friday.

**Pull request merged**